app/generic/verifier/verifier.py

```diff
@@ -27,13 +31,129 @@ class VerifierFacade(Enum):
     v20 = VerifierV2
 
 
-def __get_verifier_by_version(protocol_version: str) -> Verifier:
-    if protocol_version == "v1":
+def __get_verifier_by_version(version_candidate: str) -> Verifier:
+    if version_candidate == "v1" or version_candidate.startswith("v1-"):
         return VerifierFacade.v10.value
-    elif protocol_version == "v2":
+    elif version_candidate == "v2" or version_candidate.startswith("v2-"):
         return VerifierFacade.v20.value
     else:
-        raise ValueError(f"Unknown protocol version {protocol_version}")
+        raise ValueError(f"Unknown protocol version {version_candidate}")
+
+
+@router.post("/credentials")
+async def get_credentials_for_request(
+    proof_request: ProofRequestGeneric,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> List[IndyCredPrecis]:
+    """
+     Get matching credentials for  presentation exchange
+
+     Parameters:
+     ----------
+    proof_request: ProofRequestGeneric
+         The proof request object
+
+     Returns:
+     --------
+     presentation_exchange_list: [IndyCredPrecis]
+         The list of Indy presentation credentials
+    """
+    try:
+        prover = __get_verifier_by_version(
+            version_candidate=proof_request.protocol_version
+        )
+        return await prover.get_credentials_for_request(
+            controller=aries_controller, pres_ex_id=proof_request.proof_id
+        )
+    except Exception as e:
+        logger.error(
+            f"Failed to get matching credentials: \n {proof_request.proof_id}\n \n{e!r}"
+        )
+        raise e from e
+
+
+@router.post("/get-records")
```

Comment:
Sorry to be so pedantic again, but I think having the api just `router.get('/proofs')` is more in line with REST principles (https://jsonapi.org) and is also consistent with issuer. Then for `get_proof_record` we do `router.get('/proofs/{proof_id}')`.

If you want to go the `/get-records` and `/get-record` route you can probably convince me, but we should also change the issuer endpoint then.

I think we should put consistency as a first principle in the cloud api, As consistency makes things easy to use.

Reply:
no need to be sorry. makes it better. I'm on it.

Reply:
the only hting I'm thinking now is that for the get all with get('/proofs') how are we gonna specify the protocol version?

Reply:
i guess we could do sth like `get('/proofs/{version}'` bu tthat would make it inconsistent with what we've done so far. Maybe I'm not seeing the obvious...

Reply:
never mind actually got an idea

Reply:
In the issuer I didn't include a version specifier. But if we want, we could add it through query parameters like filters

Reply:
I've done as you've done it in issuer for now. I guess we could also return a custom type/class with v1: [v1_proofs] and v2: [v2_proofs]

Reply:
Hmm I would try to stay away from that as the protocol version being used is not really important IMO. As an entity using this API I want to request a proof. Whether that is done using v1 or v2 can have the exact same outcome. So if we can abstract that complexity that is a great advantage IMO